corehq/apps/hqwebapp/utils.py

```diff
@@ -81,3 +87,101 @@ def aliased_language_name(lang_code):
             if code == lang_code:
                 return name
         raise KeyError('Unknown language code %s' % lang_code)
+
+
+def extract_password(password):
+    # Passwords set with expected salts length and padding would respect this regex
+    reg_exp = r"^sha256\$([a-z|0-9|A-Z]{6})(.*)([a-z|0-9|A-Z]{6})=$"
```

Comment:
thanks for the correction with the `|` character.

Replacing `(.*)` with `([a-zA-Z0-9=]*)`.  `=` comes in coz the second salt used always has `=` in it at the end. Was there a specific reason for adding `+\`?